From: emre <emre@local>
Subject: [PATCH] be10000: enable USB host power

Use the upstream GL-BE10000 USB host power wiring: GPIO 23 controls
VBUS through a fixed regulator, and the board includes kmod-usb3 for
the MediaTek xHCI controller.

---
 .../mediatek/dts/mt7987a-glinet-gl-be10000.dts | 25 ++++++++--------------
 target/linux/mediatek/image/filogic.mk        |  2 +-
 2 files changed, 10 insertions(+), 17 deletions(-)

diff --git a/target/linux/mediatek/dts/mt7987a-glinet-gl-be10000.dts b/target/linux/mediatek/dts/mt7987a-glinet-gl-be10000.dts
index c05a692604..f8dd958bea 100644
--- a/target/linux/mediatek/dts/mt7987a-glinet-gl-be10000.dts
+++ b/target/linux/mediatek/dts/mt7987a-glinet-gl-be10000.dts
@@ -36,22 +36,14 @@
 		};
 	};
 
-	gpio-export {
-		compatible = "gpio-export";
-
-		usb_power {
-			gpio-export,name = "usb_power";
-			gpio-export,output = <0>;
-			gpios = <&pio 23 GPIO_ACTIVE_HIGH>;
-		};
-	};
-
-	usb-control {
-		compatible = "usb-control";
-
-		control-gpio = <&pio 24 GPIO_ACTIVE_HIGH>;
-		usb-power-path = "/sys/class/gpio/usb_power/value";
-		power-gpio-default-output = <1>;
+	usb_vbus: regulator-usb-vbus {
+		compatible = "regulator-fixed";
+		enable-active-high;
+		gpios = <&pio 23 GPIO_ACTIVE_HIGH>;
+		regulator-name = "usb_vbus";
+		regulator-boot-on;
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
 	};
 
 	fan_3v: regulator-fan-3p3v {
@@ -352,6 +344,7 @@
 };
 
 &ssusb {
+	vbus-supply = <&usb_vbus>;
 	status = "okay";
 };
 
diff --git a/target/linux/mediatek/image/filogic.mk b/target/linux/mediatek/image/filogic.mk
index 5cf1157c88..aab32b985a 100644
--- a/target/linux/mediatek/image/filogic.mk
+++ b/target/linux/mediatek/image/filogic.mk
@@ -1802,7 +1802,7 @@ define Device/glinet_gl-be10000
   DEVICE_DTS_DIR := ../dts
   DEVICE_DTC_FLAGS := --pad 4096
   DEVICE_DTS_LOADADDR := 0x4ff00000
-  DEVICE_PACKAGES := mt7987-2p5g-phy-firmware kmod-mt7992-firmware kmod-mt7996-firmware kmod-mt7996-233-firmware kmod-hwmon-pwmfan
+  DEVICE_PACKAGES := mt7987-2p5g-phy-firmware kmod-mt7992-firmware kmod-mt7996-firmware kmod-mt7996-233-firmware kmod-hwmon-pwmfan kmod-usb3
   UBINIZE_OPTS := -E 5
   BLOCKSIZE := 256k
   PAGESIZE := 4096
-- 
2.54.0
